### PR TITLE
Broadcast a client's region line color on room join

### PR DESCRIPTION
**Pull request by MikeIsAStar**

### CODE REVIEW

File: payload/game/ui/FriendMatchingPage.hh [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #pragma once
 
+#include "game/ui/ctrl/CtrlMenuDummyBack.hh"
```
Comment (vabold) [inline payload/game/ui/FriendMatchingPage.hh:3]: Subfolder includes should go after direct ones.

Comment (vabold) [inline payload/game/ui/FriendMatchingPage.hh:3]: Subfolder includes should go after direct ones.

File: payload/game/ui/FriendMatchingPage.hh [hunk 2]
```diff
@@ -77,8 +77,8 @@ private:
     PageInputManager m_inputManager;
     MatchingMessageWindow m_messageWindow;
     CtrlMenuDummyBack m_dummyBack;
-    H<PageInputManager> m_onBack{ this, &FriendMatchingPage::onBack };
-    H<YesNoPage> m_onCloseConfirm{ this, &FriendMatchingPage::onCloseConfirm };
+    H<PageInputManager> m_onBack{this, &FriendMatchingPage::onBack};
+    H<YesNoPage> m_onCloseConfirm{this, &FriendMatchingPage::onCloseConfirm};
```
Comment (vabold) [inline payload/game/ui/FriendMatchingPage.hh:81]: Missing whitespace on structure initialization.

File: payload/game/ui/FriendRoomBackPage.cc [hunk 3]
```diff
@@ -186,16 +186,16 @@ void FriendRoomBackPage::pop() {
     }
 }
 
-void FriendRoomBackPage::onPlayerJoin(System::RawMii mii, u32 location, u16 latitude,
-        u16 longitude) {
+void FriendRoomBackPage::onPlayerJoin(System::RawMii mii, u32 location, u16 latitude, u16 longitude,
+        u32 regionLineColor) {
     assert(!m_queue.full());
-    m_queue.push(Join { mii, location, latitude, longitude });
+    m_queue.push(Join{mii, location, latitude, longitude, regionLineColor});
```
Comment (vabold) [inline payload/game/ui/FriendRoomBackPage.cc:192]: Missing whitespace on structure initialization.

File: payload/game/ui/FriendRoomBackPage.cc [hunk 4]
```diff
@@ -217,7 +217,7 @@ void FriendRoomBackPage::onPlayerLeave(u32 playerId) {
     if (index != m_queue.count()) {
         m_queue.remove(index);
     } else {
-        m_queue.push(Leave { playerId });
+        m_queue.push(Leave{playerId});
```
Comment (vabold) [inline payload/game/ui/FriendRoomBackPage.cc:220]: Missing whitespace on structure initialization.

File: payload/game/ui/FriendRoomBackPage.cc [hunk 5]
```diff
@@ -230,7 +230,7 @@ void FriendRoomBackPage::onReceiveComment(u32 playerId, u32 messageId) {
         }
     }
     if (commentCount < 18) {
-        m_queue.push(Comment { playerId, messageId });
+        m_queue.push(Comment{playerId, messageId});
```
Comment (vabold) [inline payload/game/ui/FriendRoomBackPage.cc:233]: Missing whitespace on structure initialization.

File: payload/game/ui/FriendRoomBackPage.cc [hunk 6]
```diff
@@ -243,7 +243,7 @@ void FriendRoomBackPage::onSettingsChange(
             break;
         }
     }
-    m_queue.push(Settings { settings });
+    m_queue.push(Settings{settings});
```
Comment (vabold) [inline payload/game/ui/FriendRoomBackPage.cc:246]: Missing whitespace on structure initialization.

File: payload/game/ui/FriendRoomBackPage.hh [hunk 7]
```diff
@@ -1,8 +1,8 @@
 #pragma once
 
+#include "game/ui/ctrl/CtrlMenuPageTitleText.hh"
```
Comment (vabold) [inline payload/game/ui/FriendRoomBackPage.hh:3]: Subfolder includes should go after direct ones.

File: payload/sp/cs/RoomServer.cc [hunk 8]
```diff
@@ -205,16 +205,17 @@ bool RoomServer::onMain(Handler &handler) {
...
 bool RoomServer::onPlayerJoin(Handler &handler, u32 clientId, const System::RawMii *mii,
-        u32 location, u16 latitude, u16 longitude,
+        u32 location, u16 latitude, u16 longitude, u32 regionLineColor,
         const std::array<u32, RoomSettings::count> &settings) {
     if (m_playerCount == 12) {
         return false;
     }
 
-    m_players[m_playerCount] = { clientId, *mii, location, latitude, longitude, 0xFFFFFFFF, {}, settings };
+    m_players[m_playerCount] = {clientId, *mii, location, latitude, longitude, regionLineColor,
```
Comment (vabold) [inline payload/sp/cs/RoomServer.cc:214]: Missing whitespace on structure initialization.

File: payload/sp/cs/RoomServer.cc [hunk 9]
```diff
@@ -251,7 +252,7 @@ bool RoomServer::onReceiveComment(u32 playerId, u32 messageId) {
     if (m_commentQueue.full()) {
         return false;
     }
-    m_commentQueue.push(Comment { playerId, messageId });
+    m_commentQueue.push(Comment{playerId, messageId});
```
Comment (vabold) [inline payload/sp/cs/RoomServer.cc:255]: Missing whitespace on structure initialization.

**Review by vabold — CHANGES_REQUESTED** [review]
Aside from formatting, LGTM.

**Review by stblr — APPROVED** [review]

**Pull request merged**